pkg/certificate/mrc_reconciler.go

```diff
@@ -1,214 +1,181 @@
-		if err := m.updateMRCComponentStatus(mrc); err != nil {
-			return err
-		}
-	} else if shouldUpdateState(mrc) {
-		if err := m.updateMRCState(mrc); err != nil {
-			return err
-		}
-	} else if m.shouldSetIssuers(mrc) {
-		log.Info().Msgf("setting new certificate issuers on the Certificate Manager for MRC %s in stage %s", mrc.GetName(), mrc.Status.State)
-		return m.setIssuers(mrc)
+	if err = m.setIssuers(mrc1, mrc2); err != nil {
+		// TODO(jaellio): set status.state to error on MRC and potentially block on responding to MRC events corresponding to MRCs with
+		// status.state set to error
+		return err
 	}
 	return nil
 }
 
-// Component statuses determine if we're ready to transition the mrc to the next state.
-// TODO(5046): add unit tests for this, and other methods in here.
-func shouldUpdateMRCComponentStatus(mrc *v1alpha2.MeshRootCertificate) bool {
-	if isTerminal(mrc) {
-		return false
+func (m *Manager) handleSingleMRC(mrc *v1alpha2.MeshRootCertificate) error {
+	if mrc.Spec.Intent != v1alpha2.ActiveIntent {
+		log.Error().Err(ErrExpectedActiveMRC).Msgf("expected single MRC with %s intent, found %s", v1alpha2.ActiveIntent, mrc.Spec.Intent)
+		return ErrExpectedActiveMRC
 	}
 
-	if len(mrc.Status.Conditions) == 0 {
-		return false
+	issuer, err := m.getCertIssuer(mrc)
+	if err != nil {
+		return err
 	}
 
-	transitionAfter := mrc.Status.TransitionAfter
-	if transitionAfter == nil {
-		return false
-	}
+	return m.updateIssuers(issuer, issuer)
+}
 
-	//The truncate is needed so we truncate the monotonic clock.
-	return transitionAfter.Truncate(0).After(time.Now())
+var validMRCIntentCombinations = map[v1alpha2.MeshRootCertificateIntent][]v1alpha2.MeshRootCertificateIntent{
+	v1alpha2.ActiveIntent: {
+		v1alpha2.PassiveIntent,
+	},
+	v1alpha2.PassiveIntent: {
+		v1alpha2.PassiveIntent,
+		v1alpha2.ActiveIntent,
+	},
 }
 
-// shouldUpdateState (and conditions).
-func shouldUpdateState(mrc *v1alpha2.MeshRootCertificate) bool {
-	// TODO(5046): determine the status to we need to be at to be able to move forward.
-	var status v1alpha2.MeshRootCertificateComponentStatus
-
-	for _, useCase := range allUseCases {
-		switch useCase {
-		case UseCaseValidatingWebhook:
-			if mrc.Status.ComponentStatuses.ValidatingWebhook != status {
-				return false
-			}
-		case UseCaseMutatingWebhook:
-			if mrc.Status.ComponentStatuses.MutatingWebhook != status {
-				return false
-			}
-		case UseCaseXDSControlPlane:
-			if mrc.Status.ComponentStatuses.XDSControlPlane != status {
-				return false
-			}
-		case UseCaseSidecar:
-			if mrc.Status.ComponentStatuses.Sidecar != status {
-				return false
-			}
-		case UseCaseBootstrap:
-			if mrc.Status.ComponentStatuses.Bootstrap != status {
-				return false
-			}
-		case UseCaseGateway:
-			if mrc.Status.ComponentStatuses.Gateway != status {
-				return false
-			}
-		}
+// ValidateMRCIntents validates the intent combination of MRCs
+func ValidateMRCIntents(mrc1, mrc2 *v1alpha2.MeshRootCertificate) error {
```

Comment:
This could be simplified further with the removal of the `deactivate` intent. The only invalid combination of intents is `active` and `active`.